openwisp_monitoring/device/base/models.py

```diff
@@ -454,6 +454,27 @@ def handle_disabled_organization(cls, organization_id):
             status='unknown'
         )
 
+    @classmethod
+    def handle_critical_metric(cls, check, **kwargs):
```

Comment:
@praptisharma28 your tests are failing because the signature of this method is wrong. The `post_save` signal along with other model signals in Django calls the signal receivers with keyword arguments. . Read the [Django documentation for the arguments sent with post_save signal](https://docs.djangoproject.com/en/5.0/ref/signals/#post-save) 

There's no `check` argument, but it called `instance` which references the model object. The test wouldn't fail like this, if you rename the `check`  argument to `instance`.  You would need to update the code logic similarly.